payroll-processor-functions/src/payroll-processor-functions/Features/Employees/EmployeesTrigger.cs

```diff
@@ -48,5 +48,32 @@ public class EmployeesTrigger
 
             return EmployeeEntity.Map.ToEmployee(employeeEntity);
         }
+
+        [FunctionName(nameof(UpdateEmployee))]
+        public async Task<ActionResult<Employee>> UpdateEmployee(
+            [HttpTrigger(AuthorizationLevel.Anonymous, "PUT", Route = "employees/{id:Guid}")] HttpRequest req,
+            [Table(Resource.Table.Employees)] CloudTable employeeTable,
+            Guid id,
+            ILogger log)
+        {
+            log.LogInformation($"Updating an employee: [{req}]");
+
+            var updateParams = await Request.Parse<EmployeeUpdate>(req);
+
+            var querier = new TableQuerier(employeeTable);
+
+            var option = await querier.GetEntity<EmployeeEntity, Employee>(
+                updateParams.Department.ToLowerInvariant(),
+                id.ToString("n"),
+                EmployeeEntity.Map.ToEmployee);
+
+            var employee = option.IfNone(() => throw new Exception($"Could not find employee [{updateParams.Department}] [{id}]"));
+
+            employee.Status = updateParams.Status;
+
+            await employeeTable.ExecuteAsync(TableOperation.InsertOrMerge(EmployeeEntity.Map.From(employee)));
```

Comment:
I think this returns the updated entity, which will have the updated `ETag` on it.

Reply:
fixed!